diesel/src/pg/mod.rs

```diff
@@ -6,14 +6,16 @@
 
 pub mod expression;
 pub mod types;
-pub mod upsert;
+#[deprecated(since = "1.4.0", note = "Use `diesel::upsert` instead")]
+pub use super::upsert;
 
 mod backend;
 mod connection;
 mod metadata_lookup;
 mod query_builder;
 pub(crate) mod serialize;
 mod transaction;
+pub mod on_constraint;
```

Comment:
just go with `pub(crate) mod on_constraint;` and reexport that module from `::upsert::on_constraint`